### PR TITLE
#9: Allow custom tar archive path

### PR DESCRIPTION
**Pull request by reynoldsalec**
Ok, finally had time to come back to this. @Justintime50 I think the existing tests (once modified) pretty much covers things, since `test_formula.py` just passes data directly to `Formula.generate_formula_data()`. We could add some sort of test in `test_app.py` that checks the case where `TRUE` is based for URL generation (which should auto-generate the path) and the case where a custom upstream URL is passed, is that something you want?

### CODE REVIEW

**Comment by Justintime50** [comment]
Yeah the problem I'm seeing is we currently check if the param is a string which is always true in this case. GitHub passes all the params of the action to the code as a string so I unfortunately have to convert true/false strings to `True/False` bools in Python to get around this problem. We may instead want to do one of the following:

1. Change the check to state `if VAR not in set(True, False):` so that it will still respect the current true/false logic
2. OR introduce 4 new params which can take in a custom path. If this option is used, you'd basically set which targets you want to True then you would specify their custom paths via these new params.

I'm not sure which of the two is better... Thoughts?

As for the tests, what I'm expecting is for a new test to be written that creates a new formula with the custom paths. The tests don't currently assert the output formula actually match expectations - it's meant more for visual inspection. Thinking through this though I'll probably adjust that here shortly to actually assert some data to protect against regressions. Then the current test output files I'd expect to remain the same since we don't want to disturb the current true/false default behavior. 

TLDR: I'm expecting a new file to be generated in the `test/formulas` folder which will occur if you create a new test and use the `_record_formula` function - other formula tests can serve as a template on how to do this.

**Comment by Justintime50** [comment]
I added better information about tests to #23, hopefully this helps clarify what I meant above about tests ^.

I'd still like to see a test where we can prove this change doesn't break the current behavior as I feel the way it's currently written will have that effect.

**Comment by Justintime50** [comment]
I'm going to convert this to a draft until the above feedback has been addressed. Happy to take another look once that is done.

**Comment by Justintime50** [comment]
I'm going to close this PR for lack of activity, feel free to reopen if you're able to address the feedback I gave earlier. You will also want to rebase against the main branch since there were many additions there since this PR went up.

The biggest hurdle to this functionality I see is how can you define a custom URL template when pieces of the URL (such as name or version number) are dynamic on each release? It's much easier to define a well-defined standard template that all releases can follow (basically backwards which is how this is currently working) so that the user doesn't have to deal with that.
